extensions/extensions-configuration-store-vault/src/test/java/com/boozallen/aissemble/configuration/dao/VaultPropertyDaoSteps.java

```diff
@@ -0,0 +1,189 @@
+    public void oneSetOfProperties() {
+        properties = getProperties();
+    }
+
+    @Then("the VaultPropertyDao writes the all properties to vault server")
+    public void theVaultPropertyDaoWritesAllPropertiesToVault() {
+        propertyDao.write(properties);
+    }
+
+    @And("the VaultPropertyDao reads all properties back from the vault server")
+    public void theVaultPropertyDaoReadsAllPropertiesBackFromVault() {
+        for (Property property : properties) {
+            assertEquals(propertyDao.read(property.getGroupName(), property.getName()), property);
+        }
+    }
+
+    @Given("ConfigLoader reads the properties at the URIs pointing to valid base and environment configurations")
+    public void configLoaderReadsThePropertiesAtURIs() {
+        baseURI = "src/test/resources/configurations/base";
+        environmentURI = "src/test/resources/configurations/example-env";
+    }
+
+    @When("the configurations are loaded")
+    public void theConfigurationsAreLoaded() {
+        properties = configLoader.loadConfigs(baseURI, environmentURI);
+    }
+
+    @Then("the ConfigLoader uses VaultPropertyDao to write to the vault server")
+    public void theConfigLoaderUsesVaultPropertyDaoToWriteToVault() {
+        configLoader.write(properties);
+    }
+
+    @And("the properties can be read from the vault server")
+    public void thePropertiesCanBeReadFromTheVaultServer() {
+        for (Property property : properties) {
+            assertEquals(propertyDao.read(property.getGroupName(), property.getName()), property);
+        }
+    }
+
+    public Set<Property> getProperties() {
+        Set<Property> properties = new HashSet<>();
+
+        properties.add(new Property("model-training-api", "AWS_ACCESS_KEY_ID", "env-access-key-id"));
+        properties.add(new Property("model-training-api", "AWS_SECRET_ACCESS_KEY", "env-secret-access-key"));
+        properties.add(new Property("data-lineage", "connector", "smallrye-kafka"));
+        properties.add(new Property("data-lineage", "serializer", "apache.StringSerializer"));
+        properties.add(new Property("data-lineage", "topic", "lineage-topic"));
+        properties.add(new Property("data-lineage", "cloud-events", "true"));
+        properties.add(new Property("data-lineage", "added-property", "example-value"));
+        properties.add(new Property("messaging", "connector", "smallrye-kafka"));
+        properties.add(new Property("messaging", "serializer", "apache.StringSerializer"));
+        properties.add(new Property("messaging", "topic", "messaging-topic"));
+
+        return properties;
+    }
+
+    private void setupVaultContainer() {
+        final Properties encryptProperties = Krausening.getInstance().getProperties("encrypt.properties");
+        final DockerImageName vaultImage = DockerImageName.parse("aiops-docker-internal.nexus.boozallencsn.com/aiops/boozallen/aissemble-vault:1.7.0-SNAPSHOT-arm64").asCompatibleSubstituteFor("vault");
```

Comment:
S: We've had a lot of issues in the past with setting up vault in this way.  On CI, the image is typically built with the architecture-specific tag, but locally it's not.  That means we end up pulling an older version of the image from the repository instead of using/testing the just-built image.  I recall specifically the python IT for vault encryption was updated very recently to account for this.

Reply:
yeah.. this is where the `unused` function supposed to be used.. forgot to replace it.